sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -775,32 +775,61 @@ class SessionCatalog(
    * Note that, if the specified database is global temporary view database, we will list global
    * temporary views.
    */
-  def listTables(db: String): Seq[TableIdentifier] = listTables(db, "*")
```

Comment:
I think it would be better to not change the semantics of the existing listTable functions. Currently
- if dbname is global_temp, it lists global temp views.
- it always lists local temp views.
It is not fitting for our use that it always returning the local views, as I think that currently would cause the local temp view in GetTables to be duplicated - returned once for every database?

How if we avoid changing the behaviour to have:
`def listTables(db: String, pattern: String, includeLocalTempViews: Boolean)` (also keeping the part about global_temp db)
and then `def listTables(db: String, pattern: String) = listTables(db, pattern, true)`
and separately `listLocalTempViews(pattern: String)`

Then
- listTables works like it works today
- global temp views + local temp views are handled by `listTables("global_temp", "%", true)`
- just global temp views by `listTables("global_temp", "%", false)`
- just local temp views by `listLocalTempViews('%')`

and we don't need `listTempViews` and `listTablesAndTempViews`?

What do you think?
also cc @gatorsmile

Reply:
Yes. Definitely, we should not change the semantics of listTable API.